inventory_management/inventory_management/doctype/stock_entry/stock_entry.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) 2024, Viny Selopal and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from frappe.query_builder.functions import Sum
+
+
+class MandatoryWarehouseMissing(frappe.ValidationError):
+    pass
+
+
+class InsufficientItems(frappe.ValidationError):
+    pass
+
+
+class StockEntry(Document):
+    def validate(self):
+        self.validate_balance_qty()
+        self.validate_mandatory_warehouses()
+
+    def validate_balance_qty(self):
+        if self.type == "Receive":
+            return
+
+        for item_row in self.stock_entry_items:
+            item_warehouse_balance = get_item_balance_for_warehouse(
+                item_row.source_warehouse, item_row.item, self.stock_entry_items
+            )
+
+            if item_warehouse_balance < item_row.qty:
+                frappe.throw(
+                    title="Error",
+                    msg="Warehouse balance lower than requested item quantity",
+                    exc=InsufficientItems,
+                )
```

Comment:
```suggestion
                frappe.throw(
                    title=_("Insufficient Balance"),
                    msg=_("Row {0}: {1} Warehouse balance lower than requested quantity for Item {2}").format(item_row.idx, frappe.bold(item_row.source_warehouse), frappe.bold(item_row.item)
                    exc=InsufficientItems,
                )
```
We can write more informative error messages. In the modified message:
- There is a concise title that tells me what the issue is even before looking at the actual message as opposed to a generic title "Error"
- The message points me to the exact row number, item, and warehouse I need to check in the table. 
- Using `frappe.bold` to stress important points in the error message.
- Also introducing translations here. Check this writeup on translations: https://frappeframework.com/docs/user/en/translations. All user-facing strings should be translatable using "_" since our framework is multilingual.

<img width="1292" alt="image" src="https://github.com/vinyselopal/InventoryManagementFrappe/assets/24353136/1568ac97-1728-46d3-b06b-f41d29adbaed">